editor/scene_tree_dock.cpp

```diff
@@ -1077,6 +1078,8 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 					undo_redo->add_do_method(node, "set_scene_file_path", "");
 					undo_redo->add_undo_method(node, "set_scene_file_path", node->get_scene_file_path());
 					_node_replace_owner(node, node, root);
+					_node_strip_signal_inheritance(node);
+					NodeDock::get_singleton()->set_node(node); // Refresh.
```

Comment:
I wonder if there is better way to refresh it.

Reply:
I wondered too, but it didn't seem like it because all of the "refreshing" logic is inside the `set_node` method. It would require moving it out...

Maybe `NodeDock::restore_last_valid_node` would work just the same?

Reply:
It's probably fine for now, but if we ever decide to add early return to `set_node()` it will stop working.
Though my biggest concern would be that this method is not used outside editor node, so either the dock is not refreshed by anything else or there is a better way.

Reply:
I was about to go with the change, but I'll keep `set_node()` then. The lack of an early return is unexpected though, yeah.

The **NodeDock** has a pretty simple bunch of methods. None of them are like `restore_last_valid_node()` or `set_node()`, so they seem like the closest thing to a refresh?